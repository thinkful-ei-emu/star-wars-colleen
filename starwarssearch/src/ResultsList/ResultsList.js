import React from "react";

export default class ResultsList extends React.Component {
  render() {
    const results = this.props.results.map((result, index) => (
      <li key={index}>{result.name}</li>
    ));
    return <ul className="results">{results}</ul>;
  }
}
