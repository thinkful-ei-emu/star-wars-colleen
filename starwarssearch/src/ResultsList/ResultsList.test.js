import React from 'react';
import ReactDOM from 'react-dom';
import ResultsList from './ResultsList';
import renderer from 'react-test-renderer'
const results = [{name: 'Anakin'}, {name: 'Wookie'}, {name: 'jabba'}]
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ResultsList results={results}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});



it('renders the UI as expected', () => {
  const tree = renderer
    .create(<ResultsList results={results}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });