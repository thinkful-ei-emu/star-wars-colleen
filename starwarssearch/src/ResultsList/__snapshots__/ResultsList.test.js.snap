// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<ul
  className="results"
>
  <li>
    Anakin
  </li>
  <li>
    Wookie
  </li>
  <li>
    jabba
  </li>
</ul>
`;
